### ISSUE TITLE
`panic!` in both branches of `if` is linted weirdly

### ISSUE THREAD

**Issue by clarfonthey** [rust-lang/rust-clippy#17740]
### Summary

Essentially, any code of the following:

```rust
if cond {
    panic!(…);
} else {
    panic!(…);
}
```

gets labelled with a `clippy::redundant-else` and turns into:

```rust
if cond {
    panic!(…);
}
panic!(…);
```

and then a `clippy::manual-assert` turns it to:

```rust
assert!(!cond, …);
panic!(…);
```

this code is a bit less clear and so we should not trigger the original `redundant-else` if the following branch is also diverging.

### Lint Name

* `redundant-else`
* `manual-assert`

### Version

```text
rustc 1.100.0-nightly (809936eac 2026-09-12)
binary: rustc
commit-hash: 809936eac66c547a5127ce1da805f0d3a6789b98
commit-date: 2026-09-12
host: x86_64-unknown-linux-gnu
release: 1.100.0-nightly
LLVM version: 23.1.1
```
